src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -813,45 +848,250 @@ public String getName() {
+        public Control[] getControls() {
+            return userDetails.getControls();
+        }
+
+        @Override
+        public String getDn() {
+            return userDetails.getDn();
+        }
+
+        @Override
+        public GrantedAuthority[] getAuthorities() {
+            return userDetails.getAuthorities();
+        }
+
+        @Override
+        public String getPassword() {
+            return userDetails.getPassword();
+        }
+
+        @Override
+        public String getUsername() {
+            return userDetails.getUsername();
+        }
+
+        @Override
+        public boolean isAccountNonExpired() {
+            return userDetails.isAccountNonExpired();
+        }
+
+        @Override
+        public boolean isAccountNonLocked() {
+            return userDetails.isAccountNonLocked();
+        }
+
+        @Override
+        public boolean isCredentialsNonExpired() {
+            return userDetails.isCredentialsNonExpired();
+        }
+
+        @Override
+        public boolean isEnabled() {
+            return userDetails.isEnabled();
+        }
+
+        public LdapUserDetails getUserDetails() {
+            return userDetails;
+        }
+
+        public String getServer() {
+            return server;
+        }
+    }
+
+    private static class DelegateLDAPUserDetailsService implements UserDetailsService {
+        private final List<LDAPUserDetailsService> delegates;
+
+        public DelegateLDAPUserDetailsService() {
+            delegates = new ArrayList<>();
+        }
+
+        public void addDelegate(LDAPUserDetailsService delegate) {
+            delegates.add(delegate);
+        }
+
+        public boolean contains(LDAPUserDetailsService delegate) {
+            return delegates.contains(delegate);
+        }
+
+        @Override
+        public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException, DataAccessException {
+            UsernameNotFoundException lastUNFE = null;
+            for (LDAPUserDetailsService delegate : delegates) {
+                try {
+                    LdapUserDetails userDetails = delegate.loadUserByUsername(username);
+                    if (userDetails instanceof DelegatedLdapUserDetails) {
+                        return userDetails;
+                    } else {
+                        return new DelegatedLdapUserDetails(userDetails, delegate.server);
+                    }
+                } catch (UsernameNotFoundException e) {
+                    lastUNFE = e;
+                } catch (DataAccessException e) {
+                    LOGGER.log(Level.WARNING, "LDAP connection " + delegate.server + " seems to be broken, will _not_ try the next configuration.", e);
+                    throw e;
+                }
+            }
+            if (lastUNFE != null) {
+                throw  lastUNFE;
+            } else {
+                throw new UserMayOrMayNotExistException("This is not intentional.", username);
```

Comment:
Yes, here I could do that, but in `authenticate` I believe I would leak information that can be used to probe the system if I directly say that the user does not exist.